arm9/source/fatfs/ff.c

```diff
@@ -3766,9 +3771,12 @@ FRESULT f_open (
 				} else
 #endif
 				{
+					DWORD tm = GET_FATTIME();
 					/* Set directory entry initial state */
 					cl = ld_clust(fs, dj.dir);			/* Get current cluster chain */
-					st_dword(dj.dir + DIR_CrtTime, GET_FATTIME());	/* Set created time */
+					st_dword(dj.dir + DIR_CrtTime, tm);	/* Set created time */
+					st_dword(dj.dir + DIR_ModTime, tm);	/* Set modified time */
+					st_word(dj.dir + DIR_LstAccDate, (WORD)(((tm) >> 16) & 0xFFFF)); /* Set last access date */
```

Comment:
Convert `(WORD)(((tm) >> 16) & 0xFFFF)` to a macro, as it is used several times.